thetis/inversion_tools.py

```diff
@@ -71,38 +199,49 @@ def __init__(self, sta_manager, output_dir='outputs', no_exports=False, real=Fal
         self.i = 0
         self.tic = None
         self.nb_grad_evals = 0
-        self.control_coeff_list = []
-        self.control_list = []
 
     def initialize(self):
         if not self.no_exports:
             if self.real:
                 raise ValueError("Exports are not supported in Real mode.")
             create_directory(self.output_dir)
             create_directory(self.output_dir + '/hdf5')
-            for i in range(len(self.control_coeff_list)):
-                self.outfiles_m.append(
-                    fd.VTKFile(f'{self.output_dir}/control_progress_{i:02d}.pvd'))
-                self.outfiles_dJdm.append(
-                    fd.VTKFile(f'{self.output_dir}/gradient_progress_{i:02d}.pvd'))
         self.initialized = True
 
-    def add_control(self, f):
+    def add_control(self, controls, mappings=None):
         """
-        Add a control field.
+        Add control(s).
 
         Can be called multiple times in case of multiparameter optimization.
 
-        :arg f: Function or Constant to be used as a control variable.
+        :param controls: Function, Constant, or list of them representing control variables.
```

Comment:
`Constant` is no longer allowed. Maybe something like: "spatially varying Function, constant Function (as generated by :func:\`domain_constant\`), or a list of constant Functions"